src/EPR.PRN.Backend.API.UnitTests/Repositories/RepositoryTests.cs

```diff
@@ -132,4 +132,73 @@ public async Task AddPrnHistory()
         var history = await context.PrnStatusHistory.Where(p => p.CreatedByUser == statusHistory.CreatedByUser).ToListAsync();
         history.Should().HaveCount(1);
     }
+
+    [TestMethod]
+    public async Task GetModifiedPrnsbyDate_ReturnsMappedPrnUpdateStatuses()
+    {
+        //Arrange
+        var fromDate = DateTime.UtcNow.AddDays(-7);
+        var toDate = DateTime.UtcNow;
+
+        var data = _fixture.CreateMany<Eprn>().ToArray();
+        data[0].PrnNumber = "PRN001";
+        data[0].LastUpdatedDate = new DateTime(2024, 11, 21);
+        data[0].PrnStatusId = 1;
+        data[0].AccreditationYear = "2023";
+
+        data[1].PrnNumber = "PRN002";
+        data[1].LastUpdatedDate = new DateTime(2024, 11, 22);
+        data[1].PrnStatusId = 1;
+        data[1].AccreditationYear = "2024";
+
+        data[0].PrnStatusId = 1;
+        data[1].PrnStatusId = 2;
+        using var context = new EprContext(_contextOptions);
+        if (await context.Database.EnsureCreatedAsync())
+        {
+            context.AddRange(data);
+            await context.SaveChangesAsync();
+        }
+
+        //Act
+        var repo = new Repository(context);
+        var result = await repo.GetModifiedPrnsbyDate(fromDate, toDate);
+
+        //Assert
+        Assert.IsNotNull(result);
+        Assert.AreEqual(2, result.Count);
+
+        var firstPrn = result[0];
+        Assert.AreEqual("PRN001", firstPrn.EvidenceNo);
+        Assert.AreEqual("2023", firstPrn.AccreditationYear);
+        Assert.AreEqual("EV-ACCEP", firstPrn.EvidenceStatusCode);
+
+        var secondPrn = result[1];
+        Assert.AreEqual("PRN002", secondPrn.EvidenceNo);
+        Assert.AreEqual("2024", secondPrn.AccreditationYear);
+        Assert.AreEqual("EV-ACANCEL", secondPrn.EvidenceStatusCode);
+    }
+
+    [TestMethod]
+    public void MapStatusCode_ReturnsCorrectValues()
+    {
+        // Arrange
+        var privateMethod = typeof(Repository)
+            .GetMethod("MapStatusCode", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);
+
+        // Act & Assert
+        var acceptedResult = privateMethod.Invoke(_repository, new object[] { EprnStatus.ACCEPTED, "2024" });
```

Comment:
Would be more readable as driven tests rather than having six separate variables

Reply:
MapStatusCode_ReturnsCorrectValues is redundant as the scenario is already covered, so removing this test